src/collapse/dtrace.rs

```diff
@@ -168,26 +169,39 @@ impl Folder {
         )
     }
 
-    fn demangle<'a>(&self, frame: &'a str) -> Cow<'a, str> {
+    // Transforms the function part of a frame with the given function.
+    fn transform_function_name<'a, F>(&self, frame: &'a str, transform: F) -> Cow<'a, str>
+    where
+        F: Fn(&'a str) -> Cow<'a, str>,
+    {
         let mut parts = frame.splitn(2, '`');
         if let (Some(pname), Some(func)) = (parts.next(), parts.next()) {
             if self.opt.includeoffset {
                 let mut parts = func.rsplitn(2, '+');
                 if let (Some(offset), Some(func)) = (parts.next(), parts.next()) {
-                    return Cow::Owned(format!(
-                        "{}`{}+{}",
-                        pname,
-                        symbolic_demangle::demangle(func),
-                        offset
-                    ));
+                    if let Cow::Owned(func) = transform(func.trim_end()) {
+                        return Cow::Owned(format!("{}`{}+{}", pname, func, offset));
+                    }
```

Comment:
I think this is missing an `else` so we don't double-evaluate `transform` with the `if let` below.